pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +28,123 @@ import (
+	// +optional
+	CohortCondition []CohortCondition `json:"cohortCondition,omitempty"`
+
+	// CohortManagerCondition is an array of current observed cohort manager conditions.
+	// +optional
+	CohortManagerCondition []ComponentCondition `json:"cohortManagerCondition,omitempty"`
+
+	// NodeCondition is an array of current observed cohort's individual nodes conditions.
+	// +optional
+	NodeCondition []CohortNodeCondition `json:"nodeCondition,omitempty"`
+}
+
+// CohortNodeCondition contains the latest status information for some or all the
+// nodes that the cohort is comprised of.
+type CohortNodeCondition struct {
+	// Name of the node. This must be a DNS_LABEL.
+	// For example: "virtual-node-1"
+	// +optional
+	Name string `json:"name"`
+
+	// Condition is an array of current observed node conditions.
+	// +optional
+	Condition []ComponentCondition `json:"condition,omitempty"`
+}
+
+// CohortCondition contains condition information for a storage cohort.
+type CohortCondition struct {
+	// Type of component condition.
+	Type ComponentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time we got an update on a given condition.
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+// ComponentCondition contains condition information for a cohort's individual component.
+type ComponentCondition struct {
+	// Type of component condition.
+	Type string `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time we got an update on a given condition.
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+type ComponentConditionType string
+
+// These are valid conditions of cohort.
+// In the future, we can add more. The current set of conditions are:
+// CohortComponentReady, CohortComponentSchedulable.
+const (
+	// CohortComponentReady means cohort component is healthy and ready to perform its task.
+	CohortComponentReady ComponentConditionType = "Ready"
+	// CohortComponentSchedulable means the cohort component is healthy and schedulable.
+	CohortComponentSchedulable ComponentConditionType = "Schedulable"
```

Comment:
This will be used in `CohortCondition` correct?

Reply:
yes.